deploy_keys_test.go

```diff
@@ -0,0 +1,325 @@
+		fmt.Fprintf(w, `{
+			"key" : "ssh-rsa AAAA...",
+			"id" : 12,
+			"title" : "My deploy key",
+			"can_push": true,
+			"created_at" : "2015-08-29T12:44:31.550Z"
+		 }`)
+	})
+
+	key := "ssh-rsa AAAA..."
+	title := "My deploy key"
+	canPush := true
+	opt := &AddDeployKeyOptions{
+		Key:     &key,
+		Title:   &title,
+		CanPush: &canPush,
+	}
+	deployKey, _, err := client.DeployKeys.AddDeployKey(5, opt)
+	if err != nil {
+		t.Errorf("DeployKey.AddDeployKey returned error: %v", err)
+	}
+
+	timeLayout := "2006-01-02T15:04:05Z07:00"
+	createdAt, err := time.Parse(timeLayout, "2015-08-29T12:44:31.550Z")
+	if err != nil {
+		t.Errorf("DeployKeys.ListAllDeployKeys returned an error while parsing time: %v", err)
+	}
+
+	want := &DeployKey{
+		Title:     title,
+		ID:        12,
+		Key:       key,
+		CanPush:   &canPush,
+		CreatedAt: &createdAt,
+	}
+	if !reflect.DeepEqual(want, deployKey) {
+		t.Errorf("DeployKeys.AddDeployKey returned %+v, want %+v", deployKey, want)
+	}
+}
+
+func TestDeleteDeployKey(t *testing.T) {
+	mux, server, client := setup(t)
+	defer teardown(server)
+
+	mux.HandleFunc("/api/v4/projects/5/deploy_keys/13", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, http.MethodDelete)
+	})
+
+	_, err := client.DeployKeys.DeleteDeployKey(5, 13)
+	if err != nil {
+		t.Errorf("Deploykeys.DeleteDeployKey returned error: %v", err)
+	}
+}
+
+func TestEnableDeployKey(t *testing.T) {
+	mux, server, client := setup(t)
+	defer teardown(server)
+
+	mux.HandleFunc("/api/v4/projects/5/deploy_keys/13/enable", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, http.MethodPost)
+		fmt.Fprintf(w, `{
+			"key" : "ssh-rsa AAAA...",
+			"id" : 12,
+			"title" : "My deploy key",
+			"created_at" : "2015-08-29T12:44:31.550Z"
+		 }`)
+	})
+
+	deployKey, _, err := client.DeployKeys.EnableDeployKey(5, 13)
+	if err != nil {
+		t.Errorf("DeployKeys.EnableDeployKey returned error: %v", err)
+	}
+
+	timeLayout := "2006-01-02T15:04:05Z07:00"
+	createdAt, err := time.Parse(timeLayout, "2015-08-29T12:44:31.550Z")
+	if err != nil {
+		t.Errorf("DeployKeys.ListAllDeployKeys returned an error while parsing time: %v", err)
+	}
+
+	want := &DeployKey{
+		ID:        12,
+		Title:     "My deploy key",
+		Key:       "ssh-rsa AAAA...",
+		CreatedAt: &createdAt,
+	}
+	if !reflect.DeepEqual(want, deployKey) {
+		t.Errorf("DeployKeys.EnableDeployKey returned %+v, want %+v", deployKey, want)
+	}
+}
+
+func TestUpdateDeployKey(t *testing.T) {
+	mux, server, client := setup(t)
+	defer teardown(server)
+
+	mux.HandleFunc("/api/v4/projects/5/deploy_keys/11", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, http.MethodPut)
+		fmt.Fprintf(w, `{
+			"id": 11,
+			"title": "New deploy key",
+			"key": "ssh-rsa AAAA...",
+			"created_at": "2015-08-29T12:44:31.550Z",
+			"can_push": true
+		 }`)
+	})
+
+	title := "New deploy key"
+	canPush := true
+	opt := &UpdateDeployKeyOptions{
+		Title:   &title,
+		CanPush: &canPush,
+	}
```

Comment:
Please use helper functions instead:
```suggestion
	opt := &UpdateDeployKeyOptions{
		Title:   String("New deploy key"),
		CanPush: Bool(true),
	}
```